### DISCUSSION TITLE
"unknown/undefined" semantic token type

### DISCUSSION THREAD

**Discussion by benblo** [microsoft/pylance-release#2592] [Enhancement]
It would be great if any token that is not recognized would be tagged with an "unknown" type, so we can configure coloring rules for them.
For example:
```python
class Test:
	def __init__(self):
		self.my_attribute = 0

test = Test()
test.my_attribute
test.doesnt_exist

print(doesnt_exist)
```
Here , using the scopes inspector, `my_attribute` is identified with the semantic token type "property".
But, `doesnt_exist` has no semantic information _at all_:
![image](https://user-images.githubusercontent.com/574187/117116397-9c0d2300-ad8e-11eb-864b-1740fe07b84a.png)

By default, `my_attribute` appears blue, while `doesnt_exist` exist appears white, but if I wanted to highlight it in red or grey or something, what semantic rule am I supposed to modify?
It feels like this goes under an "anything" umbrella; I would rather have a "known to be unknown" type.

Similarly, unknown globals are identified by the analyzer with a yellow squiggle, but they also don't have any semantic information:
![image](https://user-images.githubusercontent.com/574187/117116811-1c338880-ad8f-11eb-8160-547a1bfcbe5b.png)
I would like to be able to change their coloring, eg if analysis is suppressed somehow (for example with `"reportUndefinedVariable": "none"`).

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
